### PR TITLE
Add custom saga in action config

### PR DESCRIPTION
**Pull request by ditkachuk**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -81,7 +81,18 @@ function getAsyncReducers() {
     };
 }
 
-export { createAsyncAction, getAsyncSagas, getAsyncReducers, noParamsReducer };
+function configAsyncAction(actionCreateConfig) {
```
Comment (megazoll) [inline src/index.js]: keep the same name and check first parameter object or not

File: src/index.js [hunk 2]
```diff
@@ -53,11 +53,11 @@ const createConfigurableAction = (action, config = {}) => {
     return newAction;
 };
 
-function createAsyncAction(actionName, handler, initialState, taker, customReducer) {
+function createAsyncAction(actionName, handler, initialState, taker, customReducer, customSagaGenerator) {
     const action = createAction(actionName);
     const dataReducer = createHttpReducer(actionName, initialState, customReducer);
     const metaReducer = createMetaReducer(actionName);
-    const saga = createSaga(action, taker);
+    const saga = createSaga(customSagaGenerator)(action, taker);
```
Comment (megazoll) [inline src/index.js]: move taker to the first function

**Comment by ditkachuk** [comment]
Added `createRequestCacheGenerator`

File: README.md [hunk 3]
```diff
@@ -1,14 +1,16 @@
 # React async client
 [![Build Status](https://travis-ci.org/experium/react-async-client.svg?branch=master)](https://travis-ci.org/experium/react-async-client)
 
-#### `createAsyncAction(actionName, handler, initialState = undefined, sagaTaker = takeLatest, customReducer = defaultReducer)`
+#### `createAsyncAction(actionName, handler, initialState = undefined, sagaTaker = takeLatest, customReducer = defaultReducer || actionConfig (Object))`
```
Comment (megazoll) [inline README.md]: seems like customReducer can be actionConfig

**Comment by ditkachuk** [comment]
Fixed with asSuccess cache reducer

File: src/utils/cache.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            });
+
+            return { response };
+        } catch(error) {
+            try {
+                const { data, lastSucceedAt } = yield getCache(action);
+
+                yield put({
+                    ...asSuccess(actionFn()),
+                    payload: data,
```
Comment (megazoll) [inline src/utils/cache.js]: can we pass data to actionFn like in first success case?

File: src/utils/cache.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import { put, select } from 'redux-saga/effects';
+import compose from 'ramda/src/compose';
+import merge from 'ramda/src/merge';
+
+import { asRequest, asError, asSuccess, asLoad } from '../actionHelpers';
+import { selectActionMeta } from '../asyncHelpers';
+import { doAction } from './doAction';
+
+export const createRequestCacheGenerator = (getCache, updateCache) => {
```
Comment (megazoll) [inline src/utils/cache.js]: getCacheItem, setCacheItem

**Comment by megazoll** [comment]
bump new version

**Comment by ditkachuk** [comment]
fixed

**Pull request merged**